types/react-gallery-carousel/index.d.ts

```diff
@@ -111,35 +130,35 @@ declare namespace Carousel {
             transitionDurationMin: number;
             transitionDurationMax: number;
             widgetsHasShadow: boolean;
-            hasLeftButton: ButtonType;
-            hasRightButton: ButtonType;
-            hasMediaButton: ButtonType;
-            hasSizeButton: ButtonType;
-            hasIndexBoard: ButtonType;
-            hasDotButtons: ButtonType;
-            hasCaptions: ButtonType;
+            hasLeftButton: SmallWidgetPositions | boolean;
+            hasRightButton: SmallWidgetPositions | boolean;
+            hasMediaButton: SmallWidgetPositions | boolean;
+            hasSizeButton: SmallWidgetPositions | boolean;
+            hasIndexBoard: SmallWidgetPositions | boolean;
+            hasDotButtons: LargeWidgetPositions | boolean;
+            hasCaptions: LargeWidgetPositions | boolean;
             hasThumbnails: boolean;
-            hasLeftButtonAtMax: ButtonType;
-            hasRightButtonAtMax: ButtonType;
-            hasMediaButtonAtMax: ButtonType;
-            hasSizeButtonAtMax: ButtonType;
-            hasIndexBoardButtonAtMax: ButtonType;
-            hasDotButtonsAtMax: ButtonType;
-            hasCaptionsAtMax: ButtonType;
+            hasLeftButtonAtMax: SmallWidgetPositions | boolean;
+            hasRightButtonAtMax: SmallWidgetPositions | boolean;
+            hasMediaButtonAtMax: SmallWidgetPositions | boolean;
+            hasSizeButtonAtMax: SmallWidgetPositions | boolean;
+            hasIndexBoardButtonAtMax: SmallWidgetPositions | boolean;
+            hasDotButtonsAtMax: LargeWidgetPositions | boolean;
+            hasCaptionsAtMax: LargeWidgetPositions | boolean;
             hasThumbnailsAtMax: boolean;
             leftIcon: React.ReactNode;
             rightIcon: React.ReactNode;
             playIcon: React.ReactNode;
             pauseIcon: React.ReactNode;
-            miniIcon: React.ReactNode;
+            minIcon: React.ReactNode;
             maxIcon: React.ReactNode;
             activeIcon: React.ReactNode;
             passiveIcon: React.ReactNode;
             shouldSwipeOnMouse: boolean;
             shouldMaximizeOnClick: boolean;
             shouldMinimizeOnClick: boolean;
             shouldMinimizeOnSwipeDown: boolean;
-            onIndexChange(): void;
+            onIndexChange({ curIndex, curIndexForDisplay }: { curIndex: number; curIndexForDisplay: number }): void;
```

Comment:
Theses params were added on this [commit](https://github.com/yifaneye/react-gallery-carousel/commit/924a06bfeb30fcb1144b5f84a7236e445bd2d965?diff=split)
![image](https://user-images.githubusercontent.com/49620737/153729708-4aaed8ff-814f-451f-b973-d6790a04b157.png)


Please let me know if this is the way that you would do it.